### DISCUSSION TITLE
parallel environment

### DISCUSSION THREAD

**Discussion by torodecamargue** [nextfoam/baram#84] [General]
Hello,

I found that some cases with lot of elements sometimes are not able to use a parallel environment.

After launch parallel = x Number of cores, the message "parallel environnement applied" is displayed but nothing is created in the "case" folder. After that, when running the case I have the error

[0] --> FOAM FATAL ERROR: (openfoam-2312)
[0] C:\Users\julie\AppData\Local\Programs\BARAM\solvers\openfoam\bin\buoyantSimpleNFoam: cannot open case directory "E:/Travail_E/Baram/calcul_1/Corning_3/case/processor0"

No processor folders are created
![image](https://github.com/nextfoam/baram/assets/75060318/692524b1-9f1d-4eab-b41e-63d99a903c0f)

Best regards

Julien

**Comment by kdg5424** [comment]
Hello Julien. This is Donggyu.

Could you give your project folder as compressed?
It looks like very weird.
After checking with your project folder, I'll let you know.

Thank you

Regards

**Comment by torodecamargue** [comment]
Hello Donggyu,

The compressed file can be download from this Wetransfer link (170MB)
https://we.tl/t-cTh8YnVE80 

The STL files are attached in this mail
[STL_OVEN.zip](https://github.com/user-attachments/files/15782498/STL_OVEN.zip)

I had already see the same problem some weeks ago, in the post attached
https://github.com/nextfoam/baram/discussions/75

Best regards

Julien

**Comment by torodecamargue** [comment]
Hello,

I found that it is possible to apply the parallel environnement only after running a calculation with 1 core. Stop the calculation and apply the parallel env and sometimes it works.

Other problem:
It is not possible to use the number of cores that we want. I have 24 cores on my computer, I try with 24 20 18 16 14 12 10 and I have the following error:

Create time

job aborted:
[ranks] message

[0-6] terminated

[7-8] process exited without calling finalize

[9] terminated

---- error analysis -----

[7-8] on DESKTOP-F6GPKIL
C:\Users\julie\AppData\Local\Programs\BARAM\solvers\openfoam\bin\buoyantSimpleNFoam ended prematurely and may have crashed. exit code 3


I can run the case with only 6 cores.. Is there some limitation with the cores number?

Best regards

Julien

**Comment by kdg5424** [comment]
Hello Julien. This is Donggyu.

We don't have the limitation with the core numbers.
And I checked it with your case, but the problem didn't happen in my computer.
My computer has 16cores. So I test 12, 8, 6, 4 cores. But the problem didn't happen.
Could you chekc your mpi settings?

![image](https://github.com/nextfoam/baram/assets/133930861/14de29cd-8853-4007-a041-9b3732ede088)

![image](https://github.com/nextfoam/baram/assets/133930861/6cd51104-62ff-4552-b089-bb6d0f29964a)

And one more thing.

Your case file seems like very weird.
First, symmetry boundary doesn't seem proper. Is there any reason you made the symmetry like this?
![image](https://github.com/nextfoam/baram/assets/133930861/9645da7e-b865-4a74-8b77-abcae7de4a22)
![image](https://github.com/nextfoam/baram/assets/133930861/f423a346-a748-4b8b-b2c5-31ba347fce47)

Second, intwall should be Thermo-coupled wall. Although you don't solve the energy equation, orginally intwall and intwall_slave is the same patch. So you should make it coupled.

Regards

Donggyu

**Comment by torodecamargue** [comment]
Hello Donggyu,

About my BC choices:
First I am only interested in the flow behaviour in this case, not the temperature distribution. So the internal wall can be a noslip condition I presume ? 

The large symmetry plane is used because the oven has a symmetry plane vertically in the middle. The small symmetry plane is used because a other ovens will be stuck on the sides. See the sketch below. 
Do you think I should not use the symmetry planes ? Slip condition is better or only the full case could be considered ? If symmetry planes are not usable, which condition sloud I use for the small symmetry plane ? 

![image](https://github.com/nextfoam/baram/assets/75060318/5bb9536f-005f-4220-a758-d35522cbee1a)

About the parallel environnement:

There is 2 different problems. The first one is, as you understood, create the processors folder according to the cores number chosen. I solve this to run first, the case with 1 core, close Baram, start again Baram and I can create the processors folder. This problem is happening on 2 different computers.

The second issue is:
Once the first problem is solved, when I start the calculation I have the following error message: 

C:\Users\julie\AppData\Local\Programs\BARAM\solvers\openfoam\bin\buoyantSimpleNFoam ended prematurely and may have crashed. exit code 3

I think that it is linked to a weird random distribution domains between cores, see image below. The distribution below gives me an error.

![image](https://github.com/nextfoam/baram/assets/75060318/894a1052-fd04-4ed3-8b85-c2e5df4a13bc)

Could you try to start the calculation with high cores number ? I succeed to run this case wih only 4 cores not more.

Best regards

Julien

**Comment by torodecamargue** [comment]
Hello,

I have updated my MPI config to v10 1.3 Problems are still here

It is the biggest problem that I face on Baram. 10 tries before finding the right cores numbers, and sometimes I have to run calculation with 1 single core...

Initialization is not working very well. with 1 core yes but in parallel, sometimes yes, sometimes no.

I found Scotch have functionnalities for splitting the domain without cutting the baffles etc...
What do you think ?  

Best regards

Julien

**Comment by jiban** [comment]
Hi Julien,
I've already started looking into this issue.
It would take some time to find out the cause of your problem, yet it has priority.
I will let you know as soon as I find something.
Thank you.

-- Jake

**Comment by torodecamargue** [comment]
Thank you for your support

**Comment by kdg5424** [comment]
Hello Julien. This is Donggyu.

We found the reason why the problem happened.
The problem happened because the porous patch also decomposed. (Originally, it shouldn't be.)
So, we'll change the settings for the decomposeParDict very soon.
We're doning some tests with the porous jump and other interfaces.
So after fixing the bug, we'll let you know.

Best regards

Donggyu

**Comment by torodecamargue** [comment]
Hello Donggyu,

Thank you I hope you will find how to fix it.

Best regards

**Comment by kdg5424** [comment]
Hello Julien. Could you let us know how you generated the PD?
As Jake told you should not check **inter-region** nor **non-comformal**. 

![image](https://github.com/nextfoam/baram/assets/133930861/eb78b297-2362-4759-98e1-7d8af5195769)

**Comment by torodecamargue** [comment]
Hello Donggyu,

Yes I only choose "interface" as shown on your picture not sub-options.

Julien

**Comment by kdg5424** [comment]
Hello Julien. We found the serious problem with your case.
This picture is the original PDtop.
![image](https://github.com/nextfoam/baram/assets/133930861/6d7c1de2-0a89-4a14-b3f2-941253117f18)

And if I decompose it, it becomes like this picture.
![image](https://github.com/nextfoam/baram/assets/133930861/98a49b82-3fa0-4946-8dc9-88d5af26adf2)
As you see, lots of faces are missing.
It appears to have occured while the processor was dividing the faces.
With a small number of processors, the probability of the face being divided is low and no problem occurs, but if it is divided into many processes, the probability of the face being split is high and the probability of a problem occurring also seems to increase.

We are finding out why this problem happens. So please wait until we find the reason and solutions.
I'm very sorry for this problem.

Regards

Donggyu kim

**Comment by torodecamargue** [comment]
Hello Donggyu,

Do you find a way to improve the parallelization when interface are used ? 

Best regards

Julien

**Comment by kdg5424** [comment]
Hello Julien. I'm really sorry for the tremendous delay. I was on the bussiness trip 19th OFW in Bejing.

We found the something wrong in baramMesh and found out the solution.
Now, the problem wolud be solved with the utility called **createBaffles**.
We're implementing it as soon as possible. (I'll discuss with jake about the date when we're implementing it.)

After we release the newest version of BARAM, we'll let you know ASAP.

Thanks

Regards

Donggyu kim

**Comment by torodecamargue** [comment]
Thank you Donggyu

Do you have an idea when will be the next release ? Approximatively
